src/helpers/game_utils.ts

```diff
@@ -410,6 +413,85 @@ export async function getMultipleChoiceOptions(
     return result;
 }
 
+/**
+ * @param userID - The user ID
+ * @returns whether the player has premium status
+ */
+export async function isUserPremium(userID: string): Promise<boolean> {
+    return !!(await dbContext
+        .kmq("premium_users")
+        .where("user_id", "=", userID)
+        .andWhere("active", "=", true)
+        .first());
+}
+
+/**
+ * @param patrons - The users to grant premium membership
+ */
+export async function addPremium(patrons: Array<Patron>): Promise<void> {
+    dbContext.kmq.transaction(async (trx) => {
+        dbContext
+            .kmq("premium_users")
+            .insert(
+                patrons.map((x) => ({
+                    user_id: x.discordID,
+                    active: x.activePatron,
+                    first_subscribed: x.firstSubscribed,
+                }))
+            )
+            .onConflict("user_id")
+            .merge()
+            .transacting(trx);
+
+        dbContext
+            .kmq("badges")
+            .insert(
+                patrons.map((x) => ({
+                    user_id: x.discordID,
+                    badge_name: PATREON_SUPPORTER_BADGE,
+                }))
+            )
+            .onConflict(["user_id", "badge_name"])
+            .ignore()
+            .transacting(trx);
+    });
+}
+
+/**
+ * @param userIDs - The users to revoke premium membership from
+ */
+export async function removePremium(userIDs: string[]): Promise<void> {
+    dbContext.kmq.transaction(async (trx) => {
+        dbContext
+            .kmq("premium_users")
+            .whereIn("user_id", userIDs)
+            .update({ active: false })
+            .transacting(trx);
+
+        dbContext
+            .kmq("badges")
+            .whereIn("user_id", userIDs)
+            .andWhere("badge_name", "=", PATREON_SUPPORTER_BADGE)
+            .del()
+            .transacting(trx);
+    });
+}
+
+/**
+ * @param guildID - The guild ID
+ * @param playerID - The player ID
+ * @returns whether the current game is a premium game, or the player is premium
+ */
+export async function isPremiumRequest(
+    guildID: string,
+    playerID: string
+): Promise<boolean> {
+    return (
+        state.gameSessions[guildID]?.isPremiumGame() ??
```

Comment:
Will this cause confusion for active players if a premium player uses `,options` while he's not in an active game?

Reply:
Don't think so. The options message has a gold embed color when it's a premium request, so it should be easily differentiated

Reply:
Won't the people in the active game think their game is premium?

Reply:
Potentially, yes. Should we use a separate color to show it's a premium game, and use that in end round messages? Maybe blue?